src/components/ScmRepositoryName/ScmRepositoryName.tsx

```diff
@@ -0,0 +1,18 @@
+import { Link } from 'react-router-dom';
+
+import { parseScmRepositoryTitle } from 'utils/utils';
+
+interface IScmRepositoryNameProps {
+  internalUrl: string;
+  scmRepositoryId: string;
+}
+
+/**
+ * The SCM Repository name that lead to SCM Repository detail page.
+ *
+ * @param internalUrl - the internal url for the SCM Repository
+ * @param scmRepositoryId - id of the SCM Repository,
+ */
+export const ScmRepositoryName = ({ internalUrl, scmRepositoryId }: IScmRepositoryNameProps) => {
```

Comment:
+1

Reply:
see a similar component: [BuildName](https://github.com/project-ncl/pnc-web-ui-react/blob/main/src/components/BuildName/BuildName.tsx), I don't think we need to put `name` at the end of `ScmRepositoryName` because the main purpose of this component is to calculate the name and chances are that we will have some ScmRepositoryName without redirection link. Also abusing the keyword `Link` will make developers confused about the difference between `ScmRepositoryNameLink` and `ScmRepositoryLink`.

I made some changes to make it more clear so that we can reuse the component for non-link SCM Repository name as well

Reply:
there is follow-up discussion, see https://github.com/project-ncl/pnc-web-ui-react/pull/172#discussion_r1138691905